### ISSUE TITLE
Add `include` / `exclude` configuration option for `contentlayer/source-files`

### ISSUE THREAD

**Issue by schickling** [contentlayerdev/contentlayer#122]
Sometimes the directory structure provided via `contentDirPath` can contain files that Contentlayer should ignore while processing the "relevant" document files. In order to allow the user to configure with files Contentlayer should consider vs ignore we could introduce an `include` and/or `exclude` option in `makeSource` of `contentlayer/source-files`.

**Comment by timlrx** [comment]
How about just watching the files specified by `filePathPattern` across all defined local `defineDocumentType`?

**Comment by schickling** [comment]
> How about just watching the files specified by `filePathPattern` across all defined local `defineDocumentType`?

Interesting idea. This is very related to #123. Need to think more about this.

**Comment by schickling** [comment]
This was just released as part of the new [`0.2.1` version](https://github.com/contentlayerdev/contentlayer/releases/tag/v0.2.1). 🚀 

Feel free to comment in case you're running into any problems or open a new issue.

**Comment by ivan-kleshnin** [comment]
It is possible to differentiate paths within the same folder?

```
blog/
  index.md -- blog page
  1.md -- posts
  2.md /
```

I setup `filePathPattern: "blog/index.md",` for a `Blog` type but then I can't find a way to skip posts (except putting them into a nested folder like `blog/posts`. Because `filePathPattern` does not support array with negation marks (e.g. with some glob libs it'd look like: `filePathPattern: ["blog/*.md", "!blog/index.md"]`).

To my understanding overlapping paths are not supported so it's not possible to defined the following:

```
filePathPattern: "**/index.md",
```

and hope it somehow won't collide with `Post` (no priority table).

---

makeSource
